systemtests/src/test/java/io/enmasse/systemtest/iot/SimpleK8sDeployTest.java

```diff
@@ -47,15 +52,43 @@
         IOT_LABELS.put("component", "iot");
     }
 
+    private static File configTempFile;
     private Kubernetes client = Kubernetes.getInstance();
 
     @BeforeAll
     protected static void setup () throws Exception {
-        KubeCMDClient.createFromFile(NAMESPACE, Paths.get("templates/iot/k8s-systemtests-config.json"));
+        Map<String, String> secrets = new HashMap<>();
+        secrets.put("iot-auth-service-tls", "systemtests-iot-auth-service-tls");
+        secrets.put("iot-tenant-service-tls", "systemtests-iot-tenant-service-tls");
+        secrets.put("iot-device-registry-tls", "systemtests-iot-device-registry-tls");
+        secrets.put("iot-http-adapter-tls", "systemtests-iot-http-adapter-tls");
+        secrets.put("iot-mqtt-adapter-tls", "systemtests-iot-mqtt-adapter-tls");
+
+        IoTConfig config = new IoTConfigBuilder()
+                .withNewMetadata()
+                .withName("default")
+                .endMetadata()
+                .withNewSpec()
+                .withNewInterServiceCertificates()
+                .withNewSecretCertificatesStrategy()
+                .withCaSecretName("systemtests-iot-service-ca")
+                .withServiceSecretNames(secrets)
+                .endSecretCertificatesStrategy()
+                .endInterServiceCertificates()
+                .endSpec()
+                .build();
+
+        configTempFile = File.createTempFile("iot-config", "json");
+        new ObjectMapper().writeValue(configTempFile, config);
+
+        KubeCMDClient.createFromFile(NAMESPACE, Paths.get(configTempFile.toURI()));
```

Comment:
I think you should be able to use the kubernetes client, to create it from the Java world, instead of calling `kubectl`.

Reply:
in systemtests we use or command line clients or we use directly the api-server.
Do you mean using the api-server approach or an approach similar to this?
https://github.com/EnMasseProject/enmasse/blob/c4061432b33ad2945e98fb6d6b7c94b6d947835d/iot/iot-tenant-service/src/main/java/io/enmasse/iot/model/IoTProjects.java#L42-L43

Reply:
@famartinrh so that's not true, ve use kubeCmdClient in only few operations (mainly for testing), calling api-server just for enmasse specific stuff, but Jens talking about using fabric8 kube client, in systemtests we use this client and we have interface for it Kubernetes.java, and since it is singleton you can get it everywhere by Kubernetes.getInstance()